core/src/apps/solana/sign_tx.py

```diff
@@ -167,4 +188,85 @@ def calculate_fee(transaction: Transaction) -> int:
                 unit_price = instruction.lamports
                 is_unit_price_set = True
 
-    return int(base_fee + math.ceil(unit_price * unit_limit / 1000000))
+    priority_fee = math.ceil(unit_price * unit_limit / 1000000)
+    return base_fee, priority_fee
+
+
+def calculate_rent(transaction: Transaction) -> int:
+    """
+        Returns max rent exemption in lamports.
+
+        To estimate rent exemption from a transaction we need to go over the instructions.
+        When new accounts are created, space must be allocated for them, rent exemption value depends on that space.
+
+        There are a handful of instruction that allocate space:
+        - System program create account instruction (the space data parameter)
+        - System program create account with seed instruction (the space data parameter)
+        - System program allocate instruction (the space data parameter)
+        - System program allocate with seed instruction (the space data parameter)
+        - Associated token account program create instruction (165 bytes for Token, 165-195 for Token22)
+        - Associated token account program create idempotent instruction (165 bytes for Token, 165-195 for Token22, might not allocate)
+    loo
+        Associated token account program allocates space based on used extensions which must be enabled in the same transaction.
+        Currently, the token22 extensions aren't supported, so the max value of 195 bytes is assumed.
+        The min Token22 account size is the same as the Token one.
```

Comment:
I think the min size is actually 170B for some reason 🙈 Maybe some space is reserved for some implicit extensions.